scripts/release/asset_publish.py

```diff
@@ -403,7 +512,13 @@ def _str2bool(v: str) -> bool:
                 if asset.type == assets.AssetType.COMPONENT:
                     # load component and check if environment exists
                     component = load_component(asset.spec_with_path)
-                    if component.type == "command":
+                    if component.type == "pipeline":
+                        if not validate_and_prepare_pipeline_component(
+                            asset.spec_with_path, final_version, registry_name
+                        ):
+                            failure_list.append(asset)
+                            continue
+                    elif component.type == "command":
```

Comment:
Another place where an enum would be nice. Also applies to pipeline reference above.